### DISCUSSION TITLE
How to merge SWR Config onError and trigger onError

### DISCUSSION THREAD

**Discussion by PedroLucasSilva** [vercel/swr#2340] [Q&A]
I noticed that when I use the trigger's onError (useSWR Mutation) it replaces the onError configured in SWRConfig. Is it possible to combine the two instead? I want to use the one configured in SWRConfig for generic errors (500) and the ones configured in the trigger to handle specific errors.

**Comment by koba04** [comment]
@PedroLucasSilva 
The `onError` of the hook is merged with `SWRConfig`, so the `onError` in `SWRConfig` is never called if you set `onError` on the hook.
But you can get the `config` from `useSWRConfig()`, so you can get the `onError` in `SWRConfig` and call it in the `onError` on the hook to call the both `onError`.

```jsx
const { onError } = useSWRConfig();
const { trigger } = useSWRMutation('/api/user', updateUser, {
  onError(...args) {
    onError(...args);
    // put your custom error handling
  }
));
```
